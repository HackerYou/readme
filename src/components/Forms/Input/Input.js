import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Input = (props) => {
    const { name,
        value,
        labelText,
        labelInline,
        type,
        placeholder,
        handleChange,
        largeLabel,
        classes,
        index,
        checked,
    } = props;
    const labelClass = classNames({
        inline: labelInline,
        largeLabel,
    });
    return (<span>
        {labelText !== '' && <label htmlFor={name} className={labelClass}>{labelText}</label>}
        <input
            name={name}
            value={value}
            type={type}
            placeholder={placeholder}
            onChange={handleChange}
            className={classes}
            data-index={index}
            checked={checked}
        />
    </span>);
};

Input.defaultProps = {
    labelInline: false,
    largeLabel: false,
    placeholder: '',
    classes: '',
    labelText: '',
    handleChange: null,
    index: 0,
    checked: false,
};

Input.propTypes = {
    classes: PropTypes.string,
    labelInline: PropTypes.bool,
    value: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    labelText: PropTypes.string,
    placeholder: PropTypes.string,
    handleChange: PropTypes.func,
    largeLabel: PropTypes.bool,
    index: PropTypes.number,
    checked: PropTypes.bool,
};

export default Input;
