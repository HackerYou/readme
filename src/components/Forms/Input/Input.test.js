import React from 'react';
import { shallow, mount } from 'enzyme';
import Input from './Input';
import renderer from 'react-test-renderer';

test('It calls the function that gets passed in as a prop to handleChange when you type in text.', () => {
    const mockHandleChange = jest.fn();
    const input = shallow(<Input labelText='label text' value='test' name='test' type='' handleChange={mockHandleChange} />);
    input.find('input').simulate('change', {target: {value: 'Simon'}});
    expect(mockHandleChange).toHaveBeenCalled();
});

test('It renders the input component with an inline label if you pass the labelInline prop', () => {
    const props = {
        labelText: "Title",
        name: "title",
        type: "text",
        value: "title",
        labelInline: true,
    };
    const component = renderer.create(<Input {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});