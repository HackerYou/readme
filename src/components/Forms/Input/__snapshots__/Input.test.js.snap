// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders the input component with an inline label if you pass the labelInline prop 1`] = `
<span>
  <label
    className="inline"
    htmlFor="title"
  >
    Title
  </label>
  <input
    checked={false}
    className=""
    data-index={0}
    name="title"
    onChange={null}
    placeholder=""
    type="text"
    value="title"
  />
</span>
`;
